networkd/network_server.ml

```diff
@@ -903,13 +903,22 @@ module Bridge = struct
                 ()
             | Some (parent, vlan) ->
                 let bridge_interfaces = Sysfs.bridge_to_interfaces name in
+                let parent_bridge_interfaces =
+                  List.filter
+                    (fun n ->
+                      Astring.String.is_prefix ~affix:"eth" n
+                      || Astring.String.is_prefix ~affix:"bond" n)
+                    (Sysfs.bridge_to_interfaces parent)
+                in
                 let parent_bridge_interface =
-                  List.hd
-                    (List.filter
-                       (fun n ->
-                         Astring.String.is_prefix ~affix:"eth" n
-                         || Astring.String.is_prefix ~affix:"bond" n)
-                       (Sysfs.bridge_to_interfaces parent))
+                  match parent_bridge_interfaces with
+                  | [] ->
+                      error
+                        {|Interface for bridge parent "%s" of vlan %i not found|}
+                        parent vlan ;
+                      raise (Network_error (Bridge_does_not_exist parent))
```

Comment:
Double-check this error makes sense

Reply:
It's an interface, so `Interface_does_not_exist`.

Reply:
Well, that also doesn't quite fit, because we don't have a name for that interface :)

Reply:
Perhaps use `Internal_error`, because it is probably a bug if you hit this.